docs/extending_fuel.rst

```diff
@@ -0,0 +1,268 @@
+typically implemented by subclassing Transformer and implementing one or both
+of these methods.
+
+As an example, here's how you would double the value of some ``'features'``
+data source.
+
+>>> from fuel.transformers import Transformer
+>>>
+>>> class FeaturesDoubler(Transformer):
+...     def __init__(self, data_stream, **kwargs):
+...         super(FeaturesDoubler, self).__init__(
+...             data_stream=data_stream,
+...             produces_examples=data_stream.produces_examples,
+...             **kwargs)
+...         
+...     def transform_example(self, example):
+...         if 'features' in self.sources:
+...             example = list(example)
+...             index = self.sources.index('features')
+...             example[index] *= 2
+...             example = tuple(example)
+...         return example
+...     
+...     def transform_batch(self, batch):
+...         if 'features' in self.sources:
+...             batch = list(batch)
+...             index = self.sources.index('features')
+...             batch[index] *= 2
+...             batch = tuple(batch)
+...         return batch
+
+Most transformers you'll implement will call their superclass constructor by
+passing the data stream and declaring whether they produce examples or batches.
+Since we wish to support both batches and examples, we'll declare our output
+type to be the same as our data stream's output type.
+
+If you were to build a transformer that only works on batches, you would pass
+``produces_examples=False`` and implement only :meth:`transform_batch`. If
+anyone tried to use your transformer on an example data stream, an error would
+automatically be raised.
+
+Let's test our doubler on some dummy dataset. **Note that this implementation
+is for illustration purposes only: it is brittle and only works on numpy
+arrays.**
+
+>>> from fuel.schemes import SequentialExampleScheme, SequentialScheme
+>>> from fuel.streams import DataStream
+>>>
+>>> dataset = IndexableDataset(
+...     indexables=OrderedDict([
+...         ('features', numpy.array([1, 2, 3, 4])),
+...         ('targets', numpy.array([-1, 1, -1, 1]))]))
+>>> example_scheme = SequentialExampleScheme(examples=dataset.num_examples)
+>>> example_stream = FeaturesDoubler(
+...     data_stream=DataStream(
+...         dataset=dataset, iteration_scheme=example_scheme))
+>>> batch_scheme = SequentialScheme(
+...     examples=dataset.num_examples, batch_size=2)
+>>> batch_stream = FeaturesDoubler(
+...     data_stream=DataStream(
+...         dataset=dataset, iteration_scheme=batch_scheme))
+>>> print([example for example in example_stream.get_epoch_iterator()])
+[(2, -1), (4, 1), (6, -1), (8, 1)]
+>>> print([batch for batch in batch_stream.get_epoch_iterator()])
+[(array([2, 4]), array([-1,  1])), (array([6, 8]), array([-1,  1]))]
+
+If you think the :meth:`transform_example` and :meth:`transform_batch`
+implementations are repetitive, you're right! In cases where the example and
```

Comment:
I would refer more explicitly to the example, so that it's clear that they are not repetitive in general. Something like "You might have noticed that in this example the :meth:`transform_example` and :meth:`transform_batch` implementations are identical."